enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult.tsx

```diff
@@ -40,30 +42,43 @@ const ResultView = ({
   questionView: QuestionView;
   setQuestionView: (questionView: QuestionView) => void;
 }) => {
+  const returnToVisualization = () => {
+    setQuestionView("visualization");
+  };
+
   if (questionView === "filter") {
     return (
       <Stack>
-        <Button onClick={() => setQuestionView("visualization")}>
-          {t`Close`}
-        </Button>
-        <Filter />
+        <Button onClick={returnToVisualization}>{t`Close`}</Button>
+        <Filter
+          onApply={returnToVisualization}
+          onClear={returnToVisualization}
+        />
       </Stack>
     );
   }
 
   if (questionView === "summarize") {
-    return <Summarize onClose={() => setQuestionView("visualization")} />;
+    return (
+      <Summarize
+        onApply={returnToVisualization}
+        onClose={returnToVisualization}
+      />
+    );
   }
 
   if (questionView === "notebook") {
-    return <Notebook onClick={() => setQuestionView("visualization")} />;
+    return <Notebook onApply={returnToVisualization} />;
   }
 
   return <QuestionVisualization />;
 };
 
 export const InteractiveQuestionResult = ({
   height,
+  withTitle = false,
```

Comment:
I noticed that the type is `withTitle?: boolean`, so leaving this out would default to undefined

Reply:
I think what he means is that the two usages are:

_InteractiveQuestion.tsx (public facing interactive question)_

```
export const _InteractiveQuestion = ({
  ...
  withTitle = false,
  ...
```

 _InteractiveAdHocQuestion.tsx (interactive question from interactive dashboard)_

```
export const InteractiveAdHocQuestion = ({
  ...
  withTitle = true,
  ...
```

So it will always have a value